indico/web/templates/forms/location_widget.html

```diff
@@ -0,0 +1,537 @@
+                            }
+                        }
+                        preventGuessing = false
+                        validRoom = false;
+                    },
+                    onNavigateBefore: function(node, query, event) {
+                        if (event.keyCode == 9) {
+                            preventGuessing = true
+                        }
+                        resultsWereOpen = node.closest('.typeahead-container').hasClass('result');
+                    },
+                    onResult: function(node, query, result, resultCount, resultCountPerGroup) {
+                        node.parent().find('.just-use-option').toggleClass('active', resultCount === 0);
+                        justUseOptionSelected = resultCount === 0;
+                    }
+                }
+            });
+
+            venueInput.on('typeahead:click-on-option', function(e, data) {
+                reorderCategories(data.item_name);
+                if (hiddenData.venue_id && data.venue_id != hiddenData.venue_id && !data.room_id) {
+                    clearRoom();
+                }
+                hiddenData.venue_name = data.item_name;
+                hiddenData.venue_id = data.venue_id;
+                postSelectionActions(venueInput, data.item_name);
+                highlightOption(venueInput, data.item_name);
+                latestSelectedOption = {field: venueInput.attr('id'), value: data.item_name};
+            })
+            .on('typeahead:click-on-custom-option', function(e, data) {
+                if (hiddenData.room_id) {
+                    clearRoom();
+                }
+                delete hiddenData.venue_id;
+                hiddenData.venue_name = data.item_name;
+                postSelectionActions(venueInput, data.item_name);
+                venueInput.val(data.item_name);
+                validVenue = true;
+                justUseOptionSelected = false;
+                venueInput.trigger('typeahead:close-results-list');
+            });
+
+            roomInput.on('typeahead:click-on-custom-option', function(e, data) {
+                delete hiddenData.room_id;
+                hiddenData.room_name = data.item_name;
+                postSelectionActions(roomInput, data.item_name);
+                roomInput.val(data.item_name);
+                validRoom = true;
+                justUseOptionSelected = false;
+                roomInput.trigger('typeahead:close-results-list');
+            });
+
+
+            $('.i-location-input-field').on('typeahead:close-results-list', function() {
+                Typeahead["#" + this.id].hideLayout();
+            }).on('typeahead:open-results-list', function() {
+                Typeahead["#" + this.id].showLayout();
+            })
+            .on('focus.typeahead', function() {
+                var $this = $(this);
+                // Move cursor to the beginning
+                $this.get(0).setSelectionRange(0,0);
```

Comment:
Yeah, wrong copy paste from the internetz, but this line goes away eventually :)